internal/service/certificatesigningrequest.go

```diff
@@ -272,33 +271,45 @@ func (h *ServiceHandler) ApproveCertificateSigningRequest(ctx context.Context, r
 	if err != nil {
 		if errors.Is(err, flterrors.ErrResourceNotFound) {
 			return server.ApproveCertificateSigningRequest404JSONResponse{}, nil
-		} else {
-			return nil, err
 		}
+		return nil, err
 	}
 
 	if api.IsStatusConditionTrue(csr.Status.Conditions, api.CertificateSigningRequestDenied) {
 		return server.ApproveCertificateSigningRequest409JSONResponse{Message: "The request has already been denied"}, nil
 	}
 
-	approvedCondition := api.Condition{
-		Type:    api.CertificateSigningRequestApproved,
-		Status:  api.ConditionStatusTrue,
-		Reason:  "Approved",
-		Message: "Approved",
-	}
-	err = h.store.CertificateSigningRequest().UpdateConditions(ctx, orgId, request.Name, []api.Condition{approvedCondition})
+	signedCert, err := signApprovedCertificateSigningRequest(h.ca, csr)
 	if err != nil {
 		if errors.Is(err, flterrors.ErrResourceNotFound) {
 			return server.ApproveCertificateSigningRequest404JSONResponse{}, nil
 		}
+		if errors.Is(err, flterrors.ErrResourceVersionConflict) {
```

Comment:
Where is this returned in signApprovedCertificateSigningRequest()?

Reply:
Hm, I think this may be outdated. The error above it also may be outdated. I can remove these.